Somewhat Refactor the Build documentation (#1561)
sile-typesetter/sile: README.md

#### Nix

> The `nix` package manager is available as a standalone package manager on many platforms other than NixOS; like most Linux and BSD distributions, macOS, and even for Windows via WSL; and thus is a viable alternative way to run SILE.

Reviewer: ```suggestion
The `nix` package manager is available as a standalone package manager on many platforms other than NixOS including most Linux and BSD distributions, macOS, and even for Windows via WSL; and thus presents a viable alternative way to run SILE on most systems.
```

[Nix packages][nix] are available in both the stable and unstable channels.
Because all packages are first made available in the unstable channel and then eventually make their way into the stable channel, to get the latest stable SILE releases we recommend hitting up the unstable channel first.
You can use all the usual Nix tricks including launching a new shell with the `sile` command available or running it directly from any shell:

```console

> $ nix shell nixos-unstable#sile

Reviewer: Unfortunately there are a lot of mixups between the legacy `nix-shell` and the new easier to use `nix shell` that isn't quite stable yet. The current invocation should be:

```suggestion
$ nix shell nixpkgs/nixpkgs-unstable#sile
```

$ sile <arguments>

> $ nix run nixos-unstable#sile -- <arguments>

Reviewer: ```suggestion
$ nix run nixpkgs/nixpkgs-unstable#sile -- <arguments>
```

```

The SILE source repository is also a valid Nix [Flake][nix-flakes] which means you can run any specific version or the latest unreleased development code directly:

```console

> $ nix run github:sile-typesetter/sile/v0.30.0 -- <sile arguments>

Reviewer: ```suggestion
$ nix run github:sile-typesetter/sile/v0.14.3 -- <sile arguments>
```

## Finding Out More

Please read the [full SILE manual][doc] for more information about what SILE is and how it can help you.

> There are example documents (source and PDF) in the [examples]/ directory.

Author: the relative hyperlink for the `[examples]` is already defined in the line `#L289`, so, i don't think repetition would be good here. how about the following?

```suggestion
There are example documents (source and PDF) in the [on the SILE website][examples].
```
